ArduCopter/commands.cpp

```diff
@@ -107,7 +107,7 @@ bool Copter::far_from_EKF_origin(const Location& loc)
 {
     // check distance to EKF origin
     Location ekf_origin;
-    if (ahrs.get_origin(ekf_origin) && (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M)) {
+    if (ahrs.get_origin(ekf_origin) && (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M) && (abs(ekf_origin.alt - loc.alt) > EKF_ORIGIN_MAX_DIST_M)) {
```

Comment:
This might need to be a separate limit.

50km is a long way vertically :-)

@rmackay9 's call, 'though

Reply:
Thank you sir for pointing that out to me..Will a limit of 1Km be an ideal threshold for altitude here ??

Reply:
I do the necessary changes

Reply:
Sir, I have worked on the suggested changes. Please take a look. Thanks!